06_puppy/pfannerj/mapstore.go

```diff
@@ -0,0 +1,54 @@
+package main
+
+import "fmt"
+
+// PuppyMap a map for storing puppies in memory.
+type PuppyMap map[uint32](Puppy)
+
+// MapStore struct for the puppy map & keeps track of the last puppy ID allocated.
+type MapStore struct {
+	puppyMap PuppyMap
+	currID   uint32
+}
+
+// NewMapStore creates a new in-memory store with map intialised.
+func NewMapStore() *MapStore {
+	var m MapStore
+	m.puppyMap = make(PuppyMap)
+	return &m
+}
+
+// CreatePuppy adds a new puppy to the map store.
+func (m *MapStore) CreatePuppy(p *Puppy) (uint32, error) {
```

Comment:
I did seek some advice on this when I was doing the peer review of PR #522 because I was a bit confused and there was already a mix of solutions merged...  I will check again and make sure it's consistent

Reply:
sorry for this lack of clarity - this mostly meant is food for thought, feel free to ignore.

Reply:
I decided to go pointer-free